src/logic/basic.lean

```diff
@@ -756,8 +759,13 @@ by simp [and_comm]
   (∃ b, (∃ a, f a = b) ∧ p b) ↔ ∃ a, p (f a) :=
 ⟨λ ⟨b, ⟨a, ha⟩, hb⟩, ⟨a, ha.symm ▸ hb⟩, λ ⟨a, ha⟩, ⟨f a, ⟨a, rfl⟩, ha⟩⟩
 
-@[simp] theorem forall_eq' {a' : α} : (∀a, a' = a → p a) ↔ p a' :=
-by simp [@eq_comm _ a']
+@[simp] theorem forall_apply_eq_imp_iff {f : α → β} {p : β → Prop} :
+  (∀ a, ∀ b, f a = b → p b) ↔ (∀ a, p (f a)) :=
+⟨λ h i, h i (f i) rfl, λ h i a hi, hi ▸ h i⟩
+
+@[simp] theorem forall_apply_eq_imp_iff' {f : α → β} {p : β → Prop} :
```

Comment:
I guess it's less likely to show up, but `b = f a` could also be a pattern: `forall_eq_apply_iff`?
Do you want to add those?

Feel free to merge if the linter is happy

bors d+

Reply:
Tempting, yeah. Is there a way for me to run the simp linter without waiting for a full mathlib build? The CI is taking a very long time on this PR, thanks to touching such low-level files I guess.

Reply:
You can (temporarily) add `#lint` at the bottom of the file, but it's not completely foolproof since later `simp` lemmas may change what the simp normal form is. I guess that's not likely for stuff in `logic.basic` though.

Reply:
Given how slow the build was for this one, I think I'll come back and add the reversed-equality lemmas in a follow-up PR.